### PR TITLE
Add QueryWorkflowByID to TestWorkflowEnvironment

### PR DESCRIPTION
**Pull request by nadilas**
## What was changed:
Implements the possibility of querying child workflows by ID in the test workflow environment

## Why?
missing feature

## Closes issue: 
https://github.com/temporalio/sdk-go/issues/390

## How has this been tested?
basic unit test included

## Any docs updates needed?
I can't find detailed docs on the exposed Testing framework features and functions at this time

### CODE REVIEW

**Review by mfateev — APPROVED** [review]
Thanks a lot for the contribution!

File: internal/internal_workflow_testsuite_test.go [hunk 1]
```diff
@@ -1857,6 +1857,79 @@ func (s *WorkflowTestSuiteUnitTest) Test_QueryWorkflow() {
 	verifyStateWithQuery(stateDone)
 }
 
+func (s *WorkflowTestSuiteUnitTest) Test_QueryChildWorkflow() {
+	queryType := "state"
+	childWorkflowId := "test-query-child-workflow"
```
Comment (vitarb) [inline internal/internal_workflow_testsuite_test.go:1862]:
```suggestion
	childWorkflowID := "test-query-child-workflow"
```

File: internal/internal_workflow_testsuite_test.go [hunk 2]
```diff
@@ -1857,6 +1857,79 @@ func (s *WorkflowTestSuiteUnitTest) Test_QueryWorkflow() {
...
+		err = ExecuteActivity(ctx, testActivityHello, signalData).Get(ctx, nil)
+		if err != nil {
+			return err
+		}
+		state = stateDone
+		return err
+	}
+	workflowFn := func(ctx Context) error {
+		cwo := ChildWorkflowOptions{
+			WorkflowID:         childWorkflowId,
```
Comment (vitarb) [inline internal/internal_workflow_testsuite_test.go:1887]:
```suggestion
			WorkflowID:         childWorkflowID,
```

File: internal/internal_workflow_testsuite_test.go [hunk 3]
```diff
@@ -1857,6 +1857,79 @@ func (s *WorkflowTestSuiteUnitTest) Test_QueryWorkflow() {
...
+		}
+		return err
+	}
+
+	env := s.NewTestWorkflowEnvironment()
+	env.RegisterWorkflow(childWorkflowFn)
+	env.RegisterWorkflow(workflowFn)
+	env.RegisterActivity(testActivityHello)
+	verifyStateWithQuery := func(expected string) {
+		encodedValue, err := env.QueryWorkflowByID(childWorkflowId, queryType, "input")
```
Comment (vitarb) [inline internal/internal_workflow_testsuite_test.go:1908]:
```suggestion
		encodedValue, err := env.QueryWorkflowByID(childWorkflowID, queryType, "input")
```

File: internal/internal_workflow_testsuite_test.go [hunk 4]
```diff
@@ -1857,6 +1857,79 @@ func (s *WorkflowTestSuiteUnitTest) Test_QueryWorkflow() {
...
+		s.NotNil(encodedValue)
+		var state string
+		err = encodedValue.Get(&state)
+		s.NoError(err)
+		s.Equal("input"+expected, state)
+	}
+
+	env.RegisterDelayedCallback(func() {
+		verifyStateWithQuery(stateWaitSignal)
+		_ = env.SignalWorkflowByID(childWorkflowId, "query-signal", "hello-query")
```
Comment (vitarb) [inline internal/internal_workflow_testsuite_test.go:1919]:
```suggestion
		_ = env.SignalWorkflowByID(childWorkflowID, "query-signal", "hello-query")
```

**Pull request merged**